### PR TITLE
TRUNK-5928: Use annotations to configure PersonAddress entity

### PR DESCRIPTION
**Pull request by abertnamanya**
…o Annotations

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5928

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by sherrif10** [comment]
cc @dkayiwa  help us approve this to run against maven. Thanks

File: api/src/main/java/org/openmrs/PersonAddress.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * v. 2.0. If a copy of the MPL was not distributed with this file, You can
  * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
  * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
- *
+ * <p>
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/PersonAddress.java:6]: Do we need this paragraph?

File: api/src/main/java/org/openmrs/PersonAddress.java [hunk 2]
```diff
@@ -17,94 +17,133 @@
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.codehaus.jackson.annotate.JsonIgnore;
+import org.hibernate.annotations.Type;
 import org.openmrs.util.OpenmrsUtil;
 
+import javax.persistence.*;
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/PersonAddress.java:23]: You can use full import statements  by removing * at the end of the package

File: api/src/main/java/org/openmrs/PersonAddress.java [hunk 3]
```diff
@@ -17,94 +17,133 @@
...
-	
+
+	@Column(name = "end_date", length = 19)
+	@Temporal(TemporalType.DATE)
 	private Date endDate;
-	
+
 	// Constructors
-	
-	/** default constructor */
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/PersonAddress.java:88]: Also some un neccessary spaces  accross all the pull request would make code clean.

**Comment by tendomart** [comment]
@abertnamanya  it looks like some files are missing license headers

**Comment by abertnamanya** [comment]
Let me look into those highlighted reviews

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 4]
```diff
@@ -56,43 +56,10 @@
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
+import org.openmrs.*;
 import org.openmrs.Order.Action;
```
Comment (LuGO0) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:60]: you should add the preferred openmrsformatter to your IDE to avoid these * imports and unwanted formatting changes.

File: api/src/main/resources/hibernate.cfg.xml [hunk 5]
```diff
@@ -57,7 +57,6 @@
 		<mapping resource="org/openmrs/api/db/hibernate/Person.hbm.xml" />
 		<mapping resource="org/openmrs/api/db/hibernate/PersonAttribute.hbm.xml" />
 		<mapping resource="org/openmrs/api/db/hibernate/PersonAttributeType.hbm.xml" />
-		<mapping resource="org/openmrs/api/db/hibernate/PersonAddress.hbm.xml" />
 		<mapping resource="org/openmrs/api/db/hibernate/PersonMergeLog.hbm.xml" />
```
Comment (LuGO0) [inline api/src/main/resources/hibernate.cfg.xml:60]: you have removed this file from the hibernate.cfg.xml but havent deleted the file itself, it should be removed since it wont be used anymore.

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 6]
```diff
@@ -56,43 +56,10 @@
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
+import org.openmrs.*;
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:59]: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotusewildcardimports

Comment (abertnamanya) [reply api/src/test/java/org/openmrs/api/OrderServiceTest.java]: @dkayiwa  I have removed the wildcard imports

**Comment by sherrif10** [comment]
@abertnamanya  some build failures to look into

File: api/src/main/java/org/openmrs/PersonAddress.java [hunk 7]
```diff
@@ -19,68 +30,105 @@
...
+@Entity
+@Table(name = "person_address")
+public class PersonAddress extends BaseChangeableOpenmrsData
+	implements java.io.Serializable, Cloneable, Comparable<PersonAddress>, Address {
+
 	public static final long serialVersionUID = 343333L;
-	
-	// Fields
 
+	// Fields
```
[the remark this answers is not in the document]
Comment (tendomart) [reply api/src/main/java/org/openmrs/PersonAddress.java:46]: And i think you got some tests failing.

File: api/src/main/java/org/openmrs/PersonAddress.java [hunk 8]
```diff
@@ -19,68 +30,105 @@
 import org.codehaus.jackson.annotate.JsonIgnore;
 import org.openmrs.util.OpenmrsUtil;
 
+
+
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/PersonAddress.java:34]: you can remove the unwanted space too

File: api/src/main/java/org/openmrs/PersonAddress.java [hunk 9]
```diff
@@ -19,68 +30,105 @@
...
 	private Integer personAddressId;
 
+	@ManyToOne
+	@JoinColumn(name = "person_id")
 	private Person person;
-	
-	private Boolean preferred = false;
 
+	@Column(name = "preferred", columnDefinition = "tinyint default false")
+	private Boolean preferred;
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/PersonAddress.java:58]:
```suggestion
	@Column(name = "preferred", nullable = false)
	private Boolean preferred = false;
```

**Review by ibacher — COMMENTED** [review]
@abertnamanya The UUID error is caused because the entries in `PatientServiceTest-mergePatients.xml` for patient_appointment not having a UUID. Just [generate some UUIDs](https://www.uuidgenerator.net/) and add the `uuid` property to those entries.

File: pom.xml [hunk 10]
```diff
@@ -517,7 +517,7 @@
 			<dependency>
 				<groupId>net.sf.ehcache</groupId>
 				<artifactId>ehcache</artifactId>
-				<version>2.10.10.1.14</version>
```
Comment (dkayiwa) [inline pom.xml:520]: Is this change required by the ticket?

Comment (abertnamanya) [reply pom.xml:520]: I added that change because the maven build would show errors due to the new version of ehcache which wasn't working on my local machine

**Comment by sherrif10** [comment]
Hey @abertnamanya  are you still working on this issue.
